### PR TITLE
Add motion catalog over pdfmake (fixes #2299)

### LINKED ISSUE #2299: Motion catalog over pdfmake
By @ThomasJunk from #2288 
 
> For printing of multiple motions, one needs to implement a ContentProvdider which does the job. Idea: MotionContentProvider has a method motionContentProvider.getContent(motion, $scope, User); which generates an Array for one motion. To render multiple motions, one could iterate over the array of motions and concat the results Pseudocode: return motions.map(function(motion){ ... return motionContentProvider.getContent(motion, $scope, User); }).reduce( function(o,n){ var tmp=o.concat(n); return tmp; },[] ).

The motion catalog shall contain:

---
(Page One:)
Heading: "Motions"

(if there are any categories, a list of categories with the Prefix and Name on first page)
Heading2: "Categories"
{Prefix} {Name}

(page two if there are categories, still one if there aren't, a table of contents with the following setup)
Heading2: "Table of Contents"
"Motion" {if set: the Category prefix} {Number}: {title} {space or dots to end of page} {page number}

---
Followed by the printable versions of the motions were every Motion is on a new page.

The page number shall be absolute to the number of pages of the catalog and not be relative to the motion.

The differences to the current ReportLab implementation are the page numbers in the toc as well as the Heading "Table of Contents"

If this is not the case or there are other requirements correct me here

@emanuelschuetze 
@normanjaeckel

### PR DESCRIPTION
**Pull request by tsiegleauq**
This needed more changes then I originally expected. I needed to change a lot of code in some factory-patterns since it was unnecessary or overly complicated in my opinion.
Also, some code was (still) misaligned.
Travis probably to refuse to check this one.
edit: After rebaseing, travis was okay with this pullrequest

@emanuelschuetze 
@normanjaeckel 

Examples:
an unfiltered catalog, just exported everyting, also had 2 categories
[sampleUnfiltered.pdf](https://github.com/OpenSlides/OpenSlides/files/490830/sampleUnfiltered.pdf)

bildschirmfoto von 2016-09-24 01-36-05

And only filtered by "Vocaloid". I removed the categories manually.
[SampleFilteredNoCat.pdf](https://github.com/OpenSlides/OpenSlides/files/490832/SampleFilteredNoCat.pdf)

### CODE REVIEW

**Comment by emanuelschuetze** [comment]
I tested your branch and get a syntax error. How to reproduce it:
* Import motions from this demo csv file: https://files.openslides.org/csv-examples/motions-demo_de.csv
* Try to export all motions as pdf catalog.
```
=>  Syntax error, unrecognized expression: Die Versammlung möge beschließen, den
Vorstand für seine letzte Legislaturperiode zu entlasten.Bericht erfolgt mündlich.
[..]
    at Scope.$scope.pdf_export (site.js:1086)
```
@tsiegleauq Please try to reproduce it with this csv examples and fix it. Thanks.

Rest works good (if I create a new motion and create pdf catalog).

**Comment by emanuelschuetze** [comment]
Please rebase and resolve conflict in site.js.

**Comment by tsiegleauq** [comment]
> I tested your branch and get a syntax error

`=>  Syntax error, unrecognized expression: ... `

This is a different issue and should be discussed in another topic.
The reason for this is that in this case we are not parsing valid HTML, since your "Die Versammlung möge beschließen [...]" is not given as valid HTML but plain text.

The parser does not care if that is valid or not, but the function to map images [introduced here](https://github.com/OpenSlides/OpenSlides/pull/2284/commits/546c4e65f64e2e350e5d73cd73c9b902914f4f62), actually does.

This is the reason why this fails:

```
var image_sources = map($(content).find("img"), function(element) {
                return element.getAttribute("src");
            });`

```
$(content) converts text to an html-document. Since a plain string is not valid HTML there will be an error.
There might be serious issues if you mig invalid HTML and regular HTML.

you should be able to reprocude this when import the CSV in current master and try to export the exact same motion

**Comment by tsiegleauq** [comment]
rebased
@emanuelschuetze

File: openslides/motions/static/js/motions/site.js [hunk 1]
```diff
@@ -1065,16 +1183,15 @@ angular.module('OpenSlidesApp.motions.site', ['OpenSlidesApp.motions', 'OpenSlid
...
     'MotionContentProvider',
     'PollContentProvider',
     'PdfMakeConverter',
     'PdfMakeDocumentProvider',
     'MotionInlineEditing',
     'gettextCatalog',
     'Projector',
-    function($scope, $http, ngDialog, MotionForm, Motion, Category, Mediafile, Tag, User, Workflow, Config,
-             motion, SingleMotionContentProvider, MotionContentProvider, PollContentProvider,
+    function($scope, $http, ngDialog, MotionComment, MotionForm, Motion, Category, Mediafile, Tag, User, Workflow, Config,
```
Comment (emanuelschuetze) [inline openslides/motions/static/js/motions/site.js]: `MotionComent` seems to be wrong here (maybe a rest of your merge conflict). Please remove it.

Comment (tsiegleauq) [inline openslides/motions/static/js/motions/site.js]: actually I inserted MotionContent manually during the rebase. I thought this was new. I just gonna remove it again.

**Comment by tsiegleauq** [comment]
@emanuelschuetze 
Removed the MotionComment and pushed. Travis is still happy, seems like there is no rebase needed this time?

**Comment by normanjaeckel** [comment]
Ok, but not tested and no review for intern pdf generation code.

**Comment by emanuelschuetze** [comment]
Retested: motion detail view works now without error. Issue for syntax error is created. So ready for merge...

**Comment by tsiegleauq** [comment]
I created the [Following Issue](https://github.com/OpenSlides/OpenSlides/issues/2433) which depends on this pull request.

**Pull request merged**